repository/repository.go

```diff
@@ -1,19 +1,29 @@
 package repository
 
 import (
+	"context"
 	"github.com/murilogilfelpeto/ps-tag-onboarding-go/configuration"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/service/models"
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
-var (
-	logger         *configuration.Logger
-	userCollection *mongo.Collection
-	context        = configuration.GetContext()
-)
+var logger = configuration.NewLogger("repository")
+
+type Repository interface {
+	Save(ctx context.Context, user models.User) (models.User, error)
+	GetUserById(ctx context.Context, id string) (models.User, error)
+	GetUserByFullName(ctx context.Context, firstName string, lastName string) (models.User, error)
+}
+
+type repository struct {
+	collection *mongo.Collection
+}
 
-func Initialize() {
-	logger = configuration.GetLogger("repository")
-	logger.Info("Initializing repository...")
+func NewRepository(client *mongo.Client, dbName string, collectionName string) Repository {
+	db := client.Database(dbName)
+	collection := db.Collection(collectionName)
```

Comment:
Consider passing the database itself instead of the client. Do you see scenarios where your app might access multiple databases? The same applies to `collectionName`, which could be defined as a constant within each data repository, that would represent something that is not expected to change over time.

Reply:
I don't see scenarios where I need more databases or collections, but I see that is UserRepository tells which database it's going to use. So if we have another repository that will use another collection or even other database, the configuration for connecting on it is done and the repository just receive the required information and how it's going to use it